### ISSUE TITLE
Combine functionality of arduino to reduce costs

### ISSUE THREAD

**Issue by Timvrakas** [tricorderproject/openct2#1]
It is (of course) up to you, but I would suggest combining some of the functionality of the 3 control boards you described into a single RPi Cape. If you include an Atmega/Attiny on the sensor array board, to do the 16 channel counting, and wire a UART/i2c/SPI interface to an RPi Cape, that also had Stepper motor drivers onboard (ex Pololu A4988 modules) you could significantly decrease the cost of the control hardware, not to mention avoiding running shielded USB cabling all over.

Just my 2 cents!
Thanks

P.S. I can't wait to build my own!

**Comment by Timvrakas** [comment]
Also, can you explain the temperature measurement system? Why are only 2 units connected?
Thanks

**Comment by Timvrakas** [comment]
I threw together a schematic with an Atmega32u4 on the array board. It automatically handles the 16 interrupts, the auto-calibration (IO expander included), and 2 ADC for the temperature probes. The board has a Micro USB connector for program/debug as well as a 4 pin (TX-RX-VLOGIC-GND) header for the Raspbery Pi Cape.

It's Here: https://github.com/Timvrakas/openct2
Thanks!

**Comment by PeterAJansen** [comment]
Hi Tim,
Sorry for taking so long to respond to this -- my notifications must have been turned off!  

I agree, the imaging system could be more compact (or at least, reduced to a fewer number of boards) by incorporating a microcontroller onto the imaging array board itself, as well as by making a raspberry pi cape with the motor controller and all the headers ready and waiting.  From a design perspective, this would certainly make things more compact, and much more integrated.  This is the approach I've usually taken in the past.

For this iteration, I've chosen to try something different for entirely personal reasons -- most people find my projects very exciting but extremely complex and challenging for the average electronics hobbyist/maker to build, and this limits how many people make them.  For this project, I've decided to try and make each piece as accessible as possible, which (here) means trading a nice Pi cape with an Arduino Uno and Motor Shield, and replacing an onboard Atmel chip on the imaging array with a through-hold header and a through-hole Arduino Pro Mini, to reduce the soldering load for those not comfortable with surface mount soldering.  The detectors are the only board with more than a few SMD parts, and each part was chosen to be in a large footprint to make it easy to solder -- nearly everything is an SOIC or 0603.  This makes it about twice as large as it needs to be, but it also makes it much more accessible and easy for folks to put together.

Thanks for putting together the schematic for the board with an integrated 32U4!  Hopefully I'll have a completed design for everything shortly, and you'll be able to start putting yours together! 

best wishes,
Peter

**Comment by Timvrakas** [comment]
Yeah I hear what you're saying about ease of assembly etc. I feel like if this thing is a successful project in the long run, its going to important to be as cost efficient as possible.

Me and my friend are planning to try and make a prototype of this at our high school tech club. Hopefully we will be able to give some useful design feedback
